sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -259,6 +262,10 @@ func newLanguageHost(exec, engineAddress, tracing, typechecker string,
 }
 
 func (host *pythonLanguageHost) connectToEngine() (pulumirpc.EngineClient, io.Closer, error) {
+	if host.engineAddress == "" {
```

Comment:
I don't think we should ever actually hit this? Can't we just error instead, make it more clear something unexpected has gone wrong.

Reply:
Done, returning an error